### PR TITLE
Refs #25614 - Always expose the provider setting

### PR DESCRIPTION
**Pull request by ekohl**
This also adds integration tests for some modules.

### CODE REVIEW

**Review by mmoll — DISMISSED** [review]
Only checked visually, but looks correct.

**Comment by tbrisker** [comment]
Pushed another commit with tests for other modules. logs and templates seem to work without the setting file expectation, seems like for some reason this method isn't called for them.

File: test/realm_freeipa/integration_test.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+require 'test_helper'
+require 'json'
+require 'root/root_v2_api'
+# require 'realm_freeipa/configuration_loader'
```
Comment (ekohl) [inline test/realm_freeipa/integration_test.rb]: Could you remove the commented code here?

**Review by ekohl — COMMENTED** [review]
I also considered testing whether it's running, but DHCP at least fails so I opted not to. If we can find an easy way to add why a plugin failed to the assertion that'd be a good test as well if we can fix it.

File: test/realm_freeipa/integration_test.rb (hunk 1 above)
Comment (tbrisker) [reply test/realm_freeipa/integration_test.rb]: oops, fixed

**Comment by tbrisker** [comment]
[test core]

**Comment by ekohl** [comment]
Could you fix the rubocop issues?

**Comment by tbrisker** [comment]
:robot: :policeman:  fixed

File: modules/templates/templates_plugin.rb [hunk 2]
```diff
@@ -6,9 +6,5 @@ class Plugin < ::Proxy::Plugin
     plugin :templates, ::Proxy::VERSION
 
     validate_presence :template_url
-
-    after_activation do
-      loading_failed "missing :foreman_url: from configuration." unless Proxy::SETTINGS.foreman_url
```
Comment (tbrisker) [inline modules/templates/templates_plugin.rb]: This is actually required. TemplatesAPI creates a [TemplateProxyRequest](https://github.com/theforeman/smart-proxy/blob/develop/modules/templates/templates_api.rb#L19), which calls [`request_factory`](https://github.com/theforeman/smart-proxy/blob/develop/modules/templates/template_proxy_request.rb#L26) which [requires this setting](https://github.com/theforeman/smart-proxy/blob/develop/lib/proxy/request.rb#L57).

File: test/templates/integration_test.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+class TemplatesApiFeaturesTest < Test::Unit::TestCase
+  include Rack::Test::Methods
+
+  def app
+    Proxy::PluginInitializer.new(Proxy::Plugins.instance).initialize_plugins
+    Proxy::RootV2Api.new
+  end
+
+  def test_features
+    # not sure why, but this module loads without loading the config file
```
Comment (tbrisker) [inline test/templates/integration_test.rb]: can this comment be removed then?

File: modules/templates/templates_plugin.rb (hunk 2 above)
Comment (ekohl) [reply modules/templates/templates_plugin.rb]: Ah, that does explain it.

**Comment by mmoll** [comment]
@ekohl which should go in first, GH-638 or this one?

**Comment by ekohl** [comment]
Rebased and squashed. @tbrisker I've added you as `Co-Authored-By` on the integration tests. I think this is a cleaner history. Should be ready to be merged.

**Review by tbrisker — APPROVED** [review]
LGTM, but since i wrote part of the code i'll leave it to @mmoll to push the button

**Comment by mmoll** [comment]
merged, bedankt @ekohl and thanks @tbrisker!

**Pull request merged**